src/App.tsx

```diff
@@ -1,204 +1,124 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useRef, useEffect, useState, useCallback } from 'react';
 import { UserWarning } from './UserWarning';
 import { USER_ID } from './api/todos';
+import * as todosServices from '../src/api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { Error } from './components/Error/Error';
+import { FilterType } from './types/FilterType';
+import cn from 'classnames';
+
+enum ErrorMessages {
+  loadTodos = 'Unable to load todos',
+  emptyTitle = 'Title should not be empty',
+  addTodo = 'Unable to add a todo',
+  deleteTodo = 'Unable to delete a todo',
+  updateTodo = 'Unable to update a todo',
+}
+
+
+const getFilteredTodos = (
+  todos: Todo[],
+  filterBy: FilterType,
+): Todo[] => {
+
+  switch (filterBy) {
+    case FilterType.active:
+      return todos.filter(todo => !todo.completed);
+    case FilterType.completed:
+      return todos.filter(todo => todo.completed);
+    default:
+      return todos;
+  }
+};
 
 export const App: React.FC = () => {
   if (!USER_ID) {
-    return <UserWarning />;
+    return (
+      <div className="todoapp">
+        <UserWarning />
```

Comment:
fix warnings.
make this condition after hooks as variant.
![Screenshot 2024-12-15 at 17 31 52](https://github.com/user-attachments/assets/99656953-495d-4987-8add-34160e00ccd3)